app/models/linked_profile.rb

```diff
@@ -31,6 +31,35 @@ class LinkedProfile < ApplicationRecord
   belongs_to :user, required: true
   belongs_to :linked_site, required: true
 
+  attr_encrypted :token, key: ENV['LP_TOKEN_KEY']
+
   validates_presence_of :url, if: :private?
   validates_presence_of :external_user_id
+  validate :verify_mal_credentials
+
+  def verify_mal_credentials
+    # Check to make sure username/password is valid
+
+    # host = MyAnimeListSyncService::ATARASHII_API_HOST
+    # request = Typhoeus::Request.new(
+    #   "#{host}account/verify_credentials",
+    #   method: :get,
+    #   userpwd: "#{external_user_id}:#{token}"
+    # )
+    # request.run
+
+    # response = request.response
+
+    # if response.code != 200
+    #   errors.add(:token, "#{response.code}: #{response.body}")
+    # end
+
+    # true
```

Comment:
Sven and I were having Webmock issues with testing it, like it flat out would not work (so that code should be valid, was just having testing issues)